src/operator.test.js

```diff
@@ -0,0 +1,109 @@
+const { ResourceEventType } = require("@dot-i/k8s-operator");
+const sinon = require("sinon");
+const { expect } = require("./test-setup");
+const Operator = require("./operator");
+const SecretHandler = require("./secret-handler");
+
+describe("operator", () => {
+  beforeEach(() => {
+    this.secretHandlerBackupMock = sinon
+      .stub(SecretHandler.prototype, "backup")
+      .resolves();
+
+    this.operator = new Operator();
+
+    this.watchResourceMock = sinon.stub(this.operator, "watchResource");
```

Comment:
Stubbing a method on the class under test does not feel right. I think it would be better to stub this method on `@dot-i/k8s-operator`